ti/instrumentation/utils.go

```diff
@@ -264,6 +267,151 @@ func getChangedFiles(ctx context.Context, workspace string, log *logrus.Logger,
+	}
+
+	// map to prevent duplicate files
+	uniqueFiles := make(map[string]struct{})
+
+	var changedRes []ti.File
+	for _, file := range res {
+		if !strings.HasSuffix(file.Name, "BUILD.bazel") {
+			// add non BUILD.bazel files to the changed files result directly and skip remainder
+			if _, exists := uniqueFiles[file.Name]; !exists {
+				changedRes = append(changedRes, file)
+				uniqueFiles[file.Name] = struct{}{}
+			}
+			continue
+		}
+		// If BUILD.Bazel at the root of harness-core is modified, run all tests /... by keeping BUILD.Bazel in changed files list
+		if strings.Replace(file.Name, "BUILD.bazel", "", -1) == "" {
+			log.Infoln("Changed file list is: ", res)
+			log.Infoln("Determined to run all tests /... : ")
+			return res, moduleList, nil
+		}
+		allJavaSrcsOutput, countAllJavaSrcs, javaTestKindOutput, err := getJavaRulesFromBazel(ctx, file.Name, workspace)
+		if err != nil {
+			if javaTestKindOutput == nil {
+				// if bazel queries fail then add module to the list to run module level tests
+				splitName := strings.Split(file.Name, "/")
+				moduleName := splitName[0]
+				moduleList = append(moduleList, moduleName)
+			}
+			continue
+		}
+		count, err := strconv.Atoi(strings.TrimSpace(string(countAllJavaSrcs)))
+		if err != nil {
+			count = 0
+		}
+		directory := filepath.Join(workspace, strings.Replace(file.Name, "/BUILD.bazel", "", -1))
+		// if no srcs present in the bazel build file, then select all files under this directory as changed
+		if count == 0 {
+			message := fmt.Sprintf("No src detected in Bazel %v, selecting all java files as changed files inside the directory %v", file.Name, directory)
+			log.Infoln(message)
+			changedRes, _ = getAllJavaFilesInsideDirectory(directory, changedRes, file, uniqueFiles)
+		} else if count >= threshold {
+			// if count of files changed due to bazel build file modification is over the limit set then update the module list with test target
+			message := fmt.Sprintf("%v Files changed after changing bazel file %v; crosses limit set for bazelFileCount", count, file.Name)
+			log.Infoln(message)
+			if javaTestKindOutput != nil {
+				splitName := strings.Split(file.Name, "/")
+				moduleName := splitName[0]
+				moduleList = append(moduleList, moduleName)
+			}
+		} else {
+			// get list of .java files from the output
+			javaFileNames := extractJavaFilesFromQueryOutput(allJavaSrcsOutput)
+			for _, name := range javaFileNames {
+				// to prevent duplicate files
+				if _, exists := uniqueFiles[name]; !exists {
+					changedRes = append(changedRes, ti.File{Status: file.Status, Name: name})
+					uniqueFiles[name] = struct{}{}
+				}
+			}
+		}
+	}
+	return changedRes, moduleList, nil
+}
+
+// takes bazelQueryOutput string and parses it to extract .java files and returns in a list
+func extractJavaFilesFromQueryOutput(bazelOutput []byte) []string {
+	var javaFileNames []string
+	// Outer Regex to extract only srcs
+	pattern1 := `srcs = \[[^\]]+\]`
```

Comment:
Does this work for cases where srcs section is split in multiple lines? Also can the `=` sign have no spaces before and after it?